### DISCUSSION TITLE
How can I publish service class?

### DISCUSSION THREAD

**Discussion by lichtner** [spatie/laravel-package-tools#96] [Q&A]
Hi there, 
my package install config file and migration, but during installation I want to also copy one Service for next customization. I need to do something like this:

```php
$this->publishes([
    __DIR__ .'/Services/MyService.php.stub' => app_path('Services/MyService.php')
]);
```

But I don't know where to put it...

How can I extend InstallCommand in MyPackageServiceProvider to publish it in the `app/Services` directory?

```php
public function configurePackage(Package $package): void
{
    $package
        ->name('laravel-my-package')
        ->hasConfigFile()
        ->hasMigration('...')
        ->hasInstallCommand(function (InstallCommand $command) {
            $command
                ->publishMigrations()
                ->publishConfigFile();
        });
}
```

I bought course, but this topic was not covered.

**Comment by freekmurze** [comment]
Take a look here: https://github.com/spatie/laravel-package-tools#working-with-a-publishable-service-provider

**Comment by frikishaan** [comment]
To publish additional files you can put the code in `packageBooted` method in your pacakge's `ServiceProvider` - 

```php

public function packageBooted(): void
{
    $this->publishes([
         __DIR__ . '/Services/MyService.php' => app_path('/Services/MyService.php'),
    ], 'package-services');
}
```
However, to publish using stubs, you have to write your own logic to replace the placeholders.

**Reply by rabol** [reply]
@lichtner  and @frikishaan  but.... this does not publish anything, simply prepare so that one can do

```php artisan vendor:publish --tag=package-services```

or do i miss something
